### PR TITLE
Add documentation about configuration file

### LINKED ISSUE #75: Update documentation regarding config files
With https://github.com/denoland/deno/pull/11776 documentation needs to be updated regarding
usage of configuration file.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #75

### CODE REVIEW

File: configuration_file.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+> specifying TypeScript options.
+
+Deno supports configuration file that allows to customize built-in TypeScript
+compiler, formatter and linter.
+
+To tell Deno to use the configuration file pass `--config path/to/file.json`
+flag.
+
+Configuration file supports `.json` and `.jsonc` extensions. We recommend to use
+`deno.json` or `deno.jsonc` as a file name, as as automatic lookup of this file
```
Comment (dsherret) [inline configuration_file.md:13]: "as as"

File: toc.json [hunk 2]
```diff
@@ -94,6 +94,9 @@
       "sanitizers": "Sanitizers"
     }
   },
+  "configuration_file": {
+    "name": "Configuration file"
+  },
```
Comment (lucacasonato) [inline toc.json:99]: I don't think this should be a top level key. Maybe part of `Getting started`, like `Command line interface`?

File: typescript/configuration.md [hunk 3]
```diff
@@ -21,6 +21,12 @@ a path on the command line. For example:
 > other TypeScript modules incompatible. Honestly it is best to use the Deno
 > defaults and to think long and hard about using a configuration file.
 
+> ⚠️ With v1.14 Deno started supporting more general configuration file, that is
```
Comment (lucacasonato) [inline typescript/configuration.md:24]:
```suggestion
> ⚠️ Deno v1.14 started supporting a more general configuration file that is
```

File: typescript/configuration.md [hunk 4]
```diff
@@ -21,6 +21,12 @@ a path on the command line. For example:
...
+> no longer confined to specyfing TypeScript compiler settings. Using
+> `tsconfig.json` as a file name will still work, but we recommend to use
+> `deno.json` or `deno.jsonc`, as automatic lookup of this file is planned for
+> the upcoming releases.
```
Comment (lucacasonato) [inline typescript/configuration.md:28]:
```suggestion
> an upcoming release.
```

File: tools/formatter.md [hunk 5]
```diff
@@ -35,3 +35,23 @@ In markdown you may use a `<!-- deno-fmt-ignore -->` comment or ignore a whole
...
+  under. Note that the printer may exceed this width in certain cases. Defaults
+  to 80.
+
+- `--options-indent-width` - The number of characters for an indent. Defaults
+  to 2.
+
+- `--options-single-quote` - Wheter to use single quote. Defaults to false
+  (using double quote).
+
+- `--options-prose-wrap` - Define how prose should be wrapped in Markdown files.
```
Comment (dsherret) [inline tools/formatter.md:56]: Should this list the possible options?

File: tools/linter.md [hunk 6]
```diff
@@ -148,3 +148,17 @@ function bar(a: any) {
...
+[a configuration file](../configuration_file.md) or following CLI flags:
+
+- `--rules-tags` - List of tag names that will be run. Empty list disables all
+  tags and will only use rules from `include`. Defaults to "recommended".
+
+- `--rules-exclude` - List of rule names that will be excluded from configured
+  tag sets. If the same rule is in `include` it be run.
+
+- `--rules-include` - List of rule names that will be run. Even if the same rule
+  is in `exclude` it will be run.
```
Comment (dsherret) [inline tools/linter.md:164]: Should we link to deno lint's website here for people to get a list of rules and tags?

File: typescript/configuration.md (hunk 3 above)
Comment (dsherret) [inline typescript/configuration.md:24]: Typo on "specyfing" and maybe move the comma over: "With v1.14, Deno started supporting a more general configuration file that is no longer confined to specifying TypeScript compiler settings."

File: typescript/configuration.md [hunk 7]
```diff
@@ -21,6 +21,12 @@ a path on the command line. For example:
 > other TypeScript modules incompatible. Honestly it is best to use the Deno
 > defaults and to think long and hard about using a configuration file.
 
+> ⚠️ With v1.14 Deno started supporting more general configuration file, that is
+> no longer confined to specyfing TypeScript compiler settings. Using
+> `tsconfig.json` as a file name will still work, but we recommend to use
+> `deno.json` or `deno.jsonc`, as automatic lookup of this file is planned for
```
Comment (dsherret) [inline typescript/configuration.md:27]: "as an automatic"

File: tools/linter.md (hunk 6 above)
Comment (bartlomieju) [reply tools/linter.md:164]: It is already linked above `For more detail about each rule, visit
[the deno_lint rule documentation](https://lint.deno.land).`

File: typescript/configuration.md [hunk 8]
```diff
@@ -21,6 +21,12 @@ a path on the command line. For example:
 > other TypeScript modules incompatible. Honestly it is best to use the Deno
 > defaults and to think long and hard about using a configuration file.
 
+> ⚠️ Deno v1.14 started supporting a more general configuration file that is no
+> longer confined to specyfing TypeScript compiler settings. Using
```
Comment (dsherret) [inline typescript/configuration.md:25]:
```suggestion
> longer confined to specifying TypeScript compiler settings. Using
```

**Review by dsherret — APPROVED** [review]

**Pull request merged**